### PR TITLE
Api forum post route

### PR DESCRIPTION
**Pull request by mrogers23458**
created an api post route for '/' in forum routes. When a post request is made to this route, it will take in title, from req.body.title, it will take in content from req.body.content, and it will take in user_id from req.session.user_id. It will then insert it into the Post model.

### CODE REVIEW

File: controllers/api/forum-routes.js [hunk 1]
```diff
@@ -1,5 +1,36 @@
 const router = require('express').Router();
-const { User } = require('../../models');
+const { User , Post } = require('../../models');
+const withAuth = require('../../utils/auth')
+
+//gets all forum posts
+router.get("/", async (req, res) => {
```
Comment (Myuze) [inline controllers/api/forum-routes.js:6]:
```suggestion
router.get('/', async (req, res) => {
```

File: controllers/api/index.js [hunk 2]
```diff
@@ -4,6 +4,10 @@ const userRoutes = require('./user-routes');
 const forumRoutes = require('./forum-routes')
 
 router.use('/users', userRoutes);
+<<<<<<< HEAD
+router.use('/forum', forumRoutes)
+=======
 router.use('/forum', forumRoutes);
+>>>>>>> dde5ae0195c07015e51679b2a1fa2a56fe2f6da9
```
Comment (Myuze) [inline controllers/api/index.js:12]: Lines 7 - 11 Unresolved Merge conflict?

File: controllers/user-routes.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            .render('new-post', { message: 'Please login or register to make a new post.'})
+        }
+    } catch (err) {
+      console.log(err);
+        res.status(500).json(err);
+    }
+  });
+
+
+module.exports = router
```
Comment (Myuze) [inline controllers/user-routes.js:31]:
```suggestion
module.exports = router

```

File: controllers/viewControllers/userViewsController.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const router = require('express').Router();
+const { User, Post } = require('../../models');
+
+router.get("/login", (req, res) => {
```
Comment (Myuze) [inline controllers/viewControllers/userViewsController.js:4]:
```suggestion
router.get('/login', (req, res) => {
```

File: models/Game.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  {
+    sequelize,
+    timestamps: false,
+    freezeTableName: true,
+    underscored: true,
+    modelName: "data",
+  }
+);
+
+module.exports = Game;
```
Comment (Myuze) [inline models/Game.js:42]:
```suggestion
module.exports = Game;

```

File: models/index.js [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
   
 Post.belongsTo(User, {
-    foreignKey: "author_id",
+    foreignKey: 'author_id',
   });
 
 User.hasOne(Data, {
-    foreignKey: "data_id",
+    foreignKey: 'data_id',
     onDelete: "CASCADE"
```
Comment (Myuze) [inline models/index.js:17]:
```suggestion
    onDelete: 'CASCADE'
```

File: models/index.js [hunk 7]
```diff
@@ -21,5 +22,14 @@ Data.belongsTo(User, {
...
+User.hasMany(Game, {
+  foreignKey: 'game_id',
+  onDelete: 'CASCADE'
+})
+
+Game.belongsTo(User, {
+  foreignKey: 'game_id'
+})
+
   module.exports = { User, Post, Data };
```
Comment (Myuze) [inline models/index.js:35]:
```suggestion
  
```

File: controllers/api/index.js (hunk 2 above)
Comment (mrogers23458) [reply controllers/api/index.js:12]: I think so, we can just delete that, and commit it with the next push.

File: controllers/api/forum-routes.js [hunk 8]
```diff
@@ -1,5 +1,36 @@
...
+    const forumData = await Post.findAll({
+      include: {
+        model: 'User',
+        attributes: [ 'id', 'username' ]
+      }
+    }).catch((err) => { 
+      res.json(err);
+    });
+      const forumPosts = forumData.map((fPost) => fPost.get({ plain: true }));
+    res.render('display', { forumPosts });
```
Comment (ENguyen25) [inline controllers/api/forum-routes.js:16]:
```suggestion
    res.render('forum', { forumPosts });
```

File: controllers/api/forum-routes.js [hunk 9]
```diff
@@ -1,5 +1,36 @@
...
+    const { user } = req.session;
+    const  forumPost = await Post.findOne({
+        where: {
+            id: req.params.id
+        }
+    }).catch((err) => { 
+      res.json(err);
+    });
+
+    res.render('display', { user, forumPost } );
```
Comment (ENguyen25) [inline controllers/api/forum-routes.js:30]:
```suggestion
    res.render('forum-post', { user, forumPost } );
```

**Review by ENguyen25 — APPROVED** [review]

**Review by Myuze — APPROVED** [review]

**Pull request merged**